### ISSUE TITLE
noborder creates transparent borders

### ISSUE THREAD

**Issue by lilydjwg** [hyprwm/Hyprland#2466]
I'm trying to remove borders for firefox nightly like this (10 for easy inspection):

```
general {
    border_size = 10
}

windowrule = noborder,^(firefox-nightly)$
```

But there is still a 10-pixel border around firefox, just transparent. (For transparent borders, one can just set border_color to a transparent color.)

I want firefox to be really noborder due to fractional scale issues. Firefox can do fraction scale by setting `widget.wayland.fractional_buffer_scale` to the scale value in `about:config`. But with scale=1.25 and border_size=1, it means Firefox window starts at (1.25, 1.25) which isn't a whole pixel, making text blurry. Setting border_size=1.25 doesn't seem to work, and setting to 4 is too large.

(This doesn't affect programs that properly support fractional scale e.g. alacritty.)

**Comment by jernejmarcic** [comment]
Regular firefox supports fractional scaling, temporarily you could try to just add a windowrule for maximizing firefox -nighly `windowule=fullscreen,^(firefox-nightly)$`

**Comment by vaxerski** [comment]
for zero pixel borders you can use `hyprctl setprop` and set the bordersize prop to 0.

**Comment by lilydjwg** [comment]
Thanks, `setprop` works, but only for one window so I need to take care to target to the window I want, like this

```sh
for add in $(hyprctl -j clients | jq -r '.[] | select(.class == "firefox-nightly") | .address'); do hyprctl setprop address:$add bordersize 0; done
```

@jernejmarcic it's 1.25x (1920x1080).

**Comment by MightyPlaza** [comment]
this is mostly fixed now on 0.28.0
